### PR TITLE
[wallet-ext] - update dapp connect accounts selector

### PR DESCRIPTION
**Pull request by mamos-mysten**
## Description 

Updates the dApp approval UI align with updated designs

https://github.com/MystenLabs/sui/assets/122397493/99e1ed95-d486-46da-b656-fd66f9009134

## Test Plan 

How did you test the new or updated feature?

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] protocol change
- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: Cargo.lock
Comment (pchrysochoidis) [inline Cargo.lock:1]: did you edit this file?

File: apps/wallet/src/ui/app/components/accounts/AccountMultiSelect.tsx
Comment (pchrysochoidis) [inline apps/wallet/src/ui/app/components/accounts/AccountMultiSelect.tsx:47]: Why do we need this? Why not use the `selectedAccountIDs` and rely on the parent updating it? I think we should use account ids in the wallet at least whenever possible

File: apps/wallet/src/ui/app/components/accounts/AccountMultiSelect.tsx
Comment (pchrysochoidis) [inline apps/wallet/src/ui/app/components/accounts/AccountMultiSelect.tsx:62]: Is this suppose to be the address or account id? There are cases that multiple accounts have the same address (rare but possible) but the id would always be unique.

File: apps/wallet/src/ui/app/components/user-approve-container/index.tsx
Comment (pchrysochoidis) [inline apps/wallet/src/ui/app/components/user-approve-container/index.tsx:87]: The button looked weird when it was transitioning from secondary to primary. Did you remove this intentionally?

File: apps/wallet/src/ui/app/pages/site-connect/index.tsx
Comment (pchrysochoidis) [inline apps/wallet/src/ui/app/pages/site-connect/index.tsx:129]: this should be account ids right? not address

File: apps/wallet/src/ui/app/pages/site-connect/index.tsx
Comment (pchrysochoidis) [inline apps/wallet/src/ui/app/pages/site-connect/index.tsx:133]: same here use account id

File: apps/wallet/src/ui/app/pages/site-connect/index.tsx
Comment (pchrysochoidis) [inline apps/wallet/src/ui/app/pages/site-connect/index.tsx:37]: We need to check if the account is unlocked before allowing the user to connect it. So we can't preselect it without checking that. We used to have a global lock but we removed it now

File: apps/wallet/src/ui/app/pages/site-connect/index.tsx
Comment (pchrysochoidis) [inline apps/wallet/src/ui/app/pages/site-connect/index.tsx:128]: Based on the design we need to split the accounts to unlocked and locked. Allow the user to connect only unlocked accounts and also have another section where user can unlock any locked accounts

File: apps/wallet/src/ui/app/pages/site-connect/index.tsx
Comment (pchrysochoidis) [inline apps/wallet/src/ui/app/pages/site-connect/index.tsx:46]: We probably want to remove duplicate addresses here. (for now maybe later we can change the permissions as well to use account ids and not addresses)

File: apps/wallet/src/ui/app/components/user-approve-container/index.tsx
Comment (mamos-mysten) [reply apps/wallet/src/ui/app/components/user-approve-container/index.tsx:87]: this was only relevant when we had a blur effect that was removed with the UI updates

File: apps/wallet/src/ui/app/components/accounts/AccountMultiSelect.tsx
Comment (mamos-mysten) [reply apps/wallet/src/ui/app/components/accounts/AccountMultiSelect.tsx:47]: The designs call for the addition of a "Select / Deselect All button" which is why we have this additional layer of state

image

File: apps/wallet/src/ui/app/components/accounts/AccountMultiSelect.tsx
Comment (mamos-mysten) [reply apps/wallet/src/ui/app/components/accounts/AccountMultiSelect.tsx:62]: updated to use account ID

File: apps/wallet/src/ui/app/pages/site-connect/index.tsx
Comment (mamos-mysten) [reply apps/wallet/src/ui/app/pages/site-connect/index.tsx:37]: I can follow up with this in the branch for locking / unlocking accounts

File: apps/wallet/src/ui/app/pages/site-connect/index.tsx
Comment (mamos-mysten) [reply apps/wallet/src/ui/app/pages/site-connect/index.tsx:50]: will follow-up in the unlock / lock accounts branch

**Comment by mamos-mysten** [comment]
added in the "Locked & Unavailable" UI: 

image

File: apps/wallet/src/ui/app/pages/site-connect/index.tsx
Comment (mamos-mysten) [reply apps/wallet/src/ui/app/pages/site-connect/index.tsx]: added this - will also update in follow-up PR for account locking / unlocking

File: apps/wallet/src/ui/app/components/accounts/AccountMultiSelect.tsx
Comment (pchrysochoidis) [reply apps/wallet/src/ui/app/components/accounts/AccountMultiSelect.tsx:53]: So we need the extra state `selectedAccounts` so show the select/deselect button? We can do the same with only `selectedAccountIDs`, right?

File: apps/wallet/src/ui/app/components/accounts/AccountsList.tsx
Comment (pchrysochoidis) [inline apps/wallet/src/ui/app/components/accounts/AccountsList.tsx:73]: I think we could move these to the component

File: apps/wallet/src/ui/app/components/accounts/AccountsList.tsx
Comment (pchrysochoidis) [inline apps/wallet/src/ui/app/components/accounts/AccountsList.tsx:34]: This better to be a mutation I think

File: apps/wallet/src/ui/app/pages/site-connect/index.tsx
Comment (pchrysochoidis) [reply apps/wallet/src/ui/app/pages/site-connect/index.tsx:41]: Lets set it to empty array then and add a ticket and a TODO

File: apps/wallet/src/ui/app/pages/site-connect/index.tsx
Comment (pchrysochoidis) [reply apps/wallet/src/ui/app/pages/site-connect/index.tsx:50]: same here

File: apps/wallet/src/ui/app/pages/site-connect/index.tsx
Comment (pchrysochoidis) [inline apps/wallet/src/ui/app/pages/site-connect/index.tsx:39]: `isLocked` means the account is locked for write. Accounts also are able to be read (un)locked. In this case if we want to be strict checking `isLocked` is fine (I am ok with this - just checking)

**Review by pchrysochoidis — APPROVED** [review]
Added a few comments that would be better to address before merging but approving to unblock

File: apps/wallet/src/ui/app/components/accounts/AccountsList.tsx
Comment (mamos-mysten) [reply apps/wallet/src/ui/app/components/accounts/AccountsList.tsx:34]: will update in the other PR

File: apps/wallet/src/ui/app/components/accounts/AccountMultiSelect.tsx
Comment (mamos-mysten) [reply apps/wallet/src/ui/app/components/accounts/AccountMultiSelect.tsx:52]: will revisit this in a future PR

**Pull request merged**
